grid_map_core/src/GridMapMath.cpp

```diff
@@ -463,105 +552,122 @@ bool incrementIndexForSubmap(Index& submapIndex, Index& index, const Index& subm
-}
-
-void getIndicesForRegions(const std::vector<Index>& regionIndeces, const Size& regionSizes,
-                          std::vector<Index> indices)
-{
-}
-
-bool colorValueToVector(const unsigned long& colorValue, Eigen::Vector3i& colorVector)
+  if (!rowMajor) {
+    return Index(
+      static_cast<int>(linearIndex) % bufferSize(
+        0), static_cast<int>(linearIndex) / bufferSize(0));
+  }
+  return Index(
+    static_cast<int>(linearIndex) / bufferSize(
+      1), static_cast<int>(linearIndex) % bufferSize(1));
+}
+
+// void getIndicesForRegion(
+//   const Index & regionIndex, const Size & regionSize,
+//   std::vector<Index> indices)
+// {
+// //  for (int i = line.index_; col < line.endIndex(); col++) {
+// //    for (int i = 0; i < getSize()(0); i++) {
+// //
+// //    }
+// //  }
+// }
+
+// void getIndicesForRegions(
+//   const std::vector<Index> & regionIndeces, const Size & regionSizes,
+//   std::vector<Index> indices)
+// {
+// }
+
+bool colorValueToVector(const uint64_t & colorValue, Eigen::Vector3i & colorVector)
 {
   colorVector(0) = (colorValue >> 16) & 0x0000ff;
   colorVector(1) = (colorValue >> 8) & 0x0000ff;
-  colorVector(2) =  colorValue & 0x0000ff;
+  colorVector(2) = colorValue & 0x0000ff;
   return true;
 }
 
-bool colorValueToVector(const unsigned long& colorValue, Eigen::Vector3f& colorVector)
+bool colorValueToVector(const uint64_t & colorValue, Eigen::Vector3f & colorVector)
 {
   Eigen::Vector3i tempColorVector;
   colorValueToVector(colorValue, tempColorVector);
   colorVector = ((tempColorVector.cast<float>()).array() / 255.0).matrix();
   return true;
 }
 
-bool colorValueToVector(const float& colorValue, Eigen::Vector3f& colorVector)
+bool colorValueToVector(const float & colorValue, Eigen::Vector3f & colorVector)
 {
   // cppcheck-suppress invalidPointerCast
-  const unsigned long tempColorValue = *reinterpret_cast<const unsigned long*>(&colorValue);
+  const uint64_t tempColorValue = *reinterpret_cast<const uint64_t *>(&colorValue);
   colorValueToVector(tempColorValue, colorVector);
   return true;
 }
 
-bool colorVectorToValue(const Eigen::Vector3i& colorVector, unsigned long& colorValue)
+bool colorVectorToValue(const Eigen::Vector3i & colorVector, uint64_t & colorValue)
 {
-  colorValue = ((int)colorVector(0)) << 16 | ((int)colorVector(1)) << 8 | ((int)colorVector(2));
+  colorValue = static_cast<int>(colorVector(0)) << 16 | static_cast<int>(colorVector(1)) << 8 |
+    static_cast<int>(colorVector(2));
   return true;
 }
 
-void colorVectorToValue(const Eigen::Vector3i& colorVector, float& colorValue)
+void colorVectorToValue(const Eigen::Vector3i & colorVector, float & colorValue)
 {
-  unsigned long color = (colorVector(0) << 16) + (colorVector(1) << 8) + colorVector(2);
-  // cppcheck-suppress invalidPointerCast
-  colorValue = *reinterpret_cast<float*>(&color);
+  uint64_t color = (colorVector(0) << 16) + (colorVector(1) << 8) + colorVector(2);
+  colorValue = static_cast<float>(color);
 }
```

Comment:
compiler was complaining about `*reinterpret_cast<float*>` replaced it with `static_cast<float>`

Reply:
Fine by me :+1: